### PR TITLE
allow configuration of the http client timeout

### PR DESCRIPTION
**Pull request by shawnkao-quadient**

### CODE REVIEW

File: data-services-client-dotnet/Api/Service.cs [hunk 1]
```diff
@@ -67,14 +67,18 @@ private void InitializeHttpClients(IConfiguration configuration)
 
             _httpClient = new HttpClient(handler)
             {
-                Timeout = TimeSpan.FromMinutes(Constants.ServiceTimeout),
+                Timeout = configuration.Timeout > 0
+                    ? TimeSpan.FromSeconds(configuration.Timeout.Value)
+                    : TimeSpan.FromMinutes(Constants.ServiceTimeout),
```
Comment (kaspersorensen) [inline data-services-client-dotnet/Api/Service.cs]: Shouldn't this also be `FromSeconds`?

File: data-services-client-dotnet/Api/Service.cs [hunk 2]
```diff
@@ -67,14 +67,18 @@ private void InitializeHttpClients(IConfiguration configuration)
...
                 BaseAddress = GetBaseAddress()
             };
             _httpClient.DefaultRequestHeaders.Add("User-Agent", configuration.UserAgent);
 
             _authClient = new HttpClient(handler)
             {
-                Timeout = TimeSpan.FromMinutes(Constants.ServiceTimeout),
+                Timeout = configuration.Timeout > 0
+                    ? TimeSpan.FromSeconds(configuration.Timeout.Value)
+                    : TimeSpan.FromMinutes(Constants.ServiceTimeout),
```
Comment (kaspersorensen) [inline data-services-client-dotnet/Api/Service.cs]: And this too?

File: data-services-client-dotnet/Api/Service.cs (hunk 1 above)
Comment (shawnkao-quadient) [reply data-services-client-dotnet/Api/Service.cs]: Doh! fixed

File: data-services-client-dotnet/Api/Service.cs (hunk 2 above)
Comment (shawnkao-quadient) [reply data-services-client-dotnet/Api/Service.cs]: fixed

**Pull request merged**